scripts/update-bcd.test.ts

```diff
@@ -821,6 +822,138 @@ describe("BCD updater", () => {
     });
   });
 
+  describe("hasSupportMatrixContradictions", () => {
+    it("detects contradictions with nonexistent support statements", () => {
+      assert.isTrue(
+        hasSupportMatrixContradictions(new Map([["80", true]]), {
+          version_added: null,
+        }),
+      );
+
+      assert.isTrue(
+        hasSupportMatrixContradictions(new Map([["80", true]]), undefined),
+      );
+    });
+
+    it("skips null support claims", () => {
+      assert.isFalse(
+        hasSupportMatrixContradictions(new Map([["80", null]]), {
+          version_added: "≤80",
+        }),
+      );
+
+      assert.isFalse(
+        hasSupportMatrixContradictions(
+          new Map([
+            ["79", false],
+            ["80", true],
+            ["81", true],
+            ["82", null],
+          ]),
+          {
+            version_added: "≤80",
+          },
+        ),
+        "skips new null test result",
+      );
+    });
+
+    it("detects contradictions in statements with boolean values", () => {
+      assert.isFalse(
+        hasSupportMatrixContradictions(new Map([["80", false]]), {
+          version_added: false,
+        }),
+        "skips generic false statements",
+      );
+      // TODO: verify this is the result that we want
+      assert.isFalse(
+        hasSupportMatrixContradictions(new Map([["80", true]]), {
+          version_added: true,
+        }),
+        "skips generic true statements",
```

Comment:
I wasn't sure how to handle situations where BCD just has a `true` value for `version_added`, so for now we're saying there is **no** **contradiction** if a test result also shows specific version support. This means we would skip updating this entry. 

It doesn't seem like this is a very common scenario at all (for ex. there are no entries in BCD currently that have `version_added: true`). Let me know if folks think we should handle this differently.

Reply:
This is definitely a case where the collector should change the results in BCD.  While it's not a _contradiction_ (both statements say the feature is supported), `version_added: true` means that we don't know what browser version a feature landed.  Since the collector _does_ know what version a feature was added in, it should update BCD with the version number.